### PR TITLE
feat(24.04): add xz-utils slice

### PR DESCRIPTION
**Pull request by endersonmaia**
# Proposed changes

Add xz-utils slice.

## Checklist

* [x] I have read the [contributing guidelines](
https://github.com/canonical/chisel-releases/blob/main/CONTRIBUTING.md)
* [x] I have tested my changes ([see how](https://github.com/canonical/chisel-releases/blob/main/CONTRIBUTING.md#7-test-your-slices-before-opening-a-pr))
* [x] I have already submitted the [CLA form](
https://ubuntu.com/legal/contributors/agreement)

### CODE REVIEW

**Comment by clay-lake** [comment]
Thanks for the contribution! The slice itself looks good to me. However could you add some tests to help us verify everything works as expected? https://github.com/canonical/chisel-releases/blob/main/CONTRIBUTING.md#7-test-your-slices-before-opening-a-pr

Here is an example PR with tests:
https://github.com/canonical/chisel-releases/pull/320/files#diff-40e68e9abb4e657ef292dcfd6fa750ddffea20daec62f7ceb9acc7605beb17e3

File: slices/xz-utils.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package: xz-utils
```
Comment (clay-lake) [inline slices/xz-utils.yaml:1]: Thanks for the contribution! The slice itself looks good to me. However could you add some spread tests to help us verify everything works as expected? https://github.com/canonical/chisel-releases/blob/main/CONTRIBUTING.md#7-test-your-slices-before-opening-a-pr

Here is an example PR with tests:
https://github.com/canonical/chisel-releases/pull/320/files#diff-40e68e9abb4e657ef292dcfd6fa750ddffea20daec62f7ceb9acc7605beb17e3

**Review by clay-lake — CHANGES_REQUESTED** [review]

File: slices/xz-utils.yaml (hunk 1 above)
Comment (endersonmaia) [reply slices/xz-utils.yaml:1]: I suppose I don't need to add a test for every binary provided.

But when testing `xz` I thought I could test with `xz --version`, `xz -z` (compress) and a little `xzgrep` just for "completeness" of the test. But this failed and I can't understand why.

I'll send a fixup without the `xzgrep` that may pass, but I am not satisfied.

File: slices/xz-utils.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+essential:
+  - xz-utils_copyright
+
+slices:
```
Comment (clay-lake) [inline slices/xz-utils.yaml:6]: It looks like xzgrep and a few of the other utils in `/usr/bin/` maybe shell scripts. Could we move the scripts to a util folder and add the dependencies there?

```
slices:
  bins:
    essential:
      - libc6_libs
      - liblzma5_libs
    contents:
      /usr/bin/lzmainfo:
      /usr/bin/unxz:
      /usr/bin/xz:
slices:
  bins:
    essential:
      - xz-utils_bins
      - dash_bins
      - grep_bins
      ... possibly more dependencies
    contents:
      /usr/bin/xzcat:
      /usr/bin/xzcmp:
      /usr/bin/xzdiff:
      /usr/bin/xzegrep:
      /usr/bin/xzfgrep:
      /usr/bin/xzgrep:
      /usr/bin/xzless:
      /usr/bin/xzmore:
```

**Review by clay-lake — CHANGES_REQUESTED** [review]

File: slices/xz-utils.yaml (hunk 2 above)
Comment (endersonmaia) [reply slices/xz-utils.yaml:6]: You mean create a new slice called utils?

```yaml
slices:
...
  utils:
    essential:
      - xz-utils_bins
      - dash_bins
      - grep_bins
```

I just think it's gonna be ugly to call `xz-utils_utils`.

Comment (endersonmaia) [reply slices/xz-utils.yaml:6]: I couldn't make it pass the test I've made, but sent the commit anyway so you can help me track what's missing.


I still get this error:

```
chroot: failed to run command ‘xzgrep’: No such file or directory
```

Comment (clay-lake) [reply slices/xz-utils.yaml:6]: I made a rootfs with your slices and busybox so I can take a closer look at the environment, and was able to recreate your error. It looks like xzgrep is expecting /bin/sh to exist. I symlinked /bin to /usr/bin and /usr/bin/sh to /usr/bin/busybox and xzgrep seems to execute now.

I think I saw another slice with a similar hurdle, let me see what the convention is for fixing this.

Comment (clay-lake) [reply slices/xz-utils.yaml:6]: Sorry for the delay. The team mentioned we need to add base-files_bin (or base-files_base) to link /bin to /usr/bin.  I gave it a shot and there were still a few broken lines in the test task. I think some of the process substitution might be breaking under chroot. Tomorrow I'll see if I can come up with an alternate test.

Comment (clay-lake) [reply slices/xz-utils.yaml:6]: Here is a draft PR with some changes I made to get it to run locally. Lets see if it passes spread tests. Feel free to cherry-pick the changes if you like them.

https://github.com/canonical/chisel-releases/pull/338

**Comment by clay-lake** [comment]
Hi @endersonmaia, I added some final changes with the team in #338. Feel free to add some comments there, otherwise I will merge it soon, and close this PR.

Thanks again for the contribution!

**Comment by linostar** [comment]
Closing this as #338 was merged.

**Pull request closed without merging**